cmd/skipper/main.go

```diff
@@ -697,7 +691,7 @@ func main() {
 		// connections, timeouts:
 		WaitForHealthcheckInterval:   waitForHealthcheckInterval,
 		IdleConnectionsPerHost:       idleConnsPerHost,
-		CloseIdleConnsPeriod:         time.Duration(clsic) * time.Second,
+		CloseIdleConnsPeriod:         closeIdleConnsPeriod,
```

Comment:
i think to fix the bug, this single change is the right one. All the rest of the changes are not required and they also would break compatibility.